system-addon/content-src/components/TopSites/TopSiteForm.jsx

```diff
@@ -123,17 +123,27 @@ export class TopSiteForm extends React.PureComponent {
     return url;
   }
 
-  validateUrl(url) {
+  _tryParseUrl(url) {
     try {
-      return !!new URL(this.cleanUrl(url));
+      return new URL(url);
     } catch (e) {
       return false;
     }
   }
 
+  validateUrl(url) {
+    const validProtocols = ["http:", "https:"];
+    const urlObj = this._tryParseUrl(url) || this._tryParseUrl(`https://${url}`);
```

Comment:
Any reason to add "https" here but "http" for `cleanUrl`? Why not just reuse `cleanUrl` for the fallback?